source/plugins/go/src/oms.go

```diff
@@ -1151,10 +1209,26 @@ func PostDataHelper(tailPluginRecords []map[interface{}]interface{}) int {
 			if containerID == "" || containsKey(StdoutIgnoreNsSet, k8sNamespace) {
 				continue
 			}
+			if containsKey(SystemNamespaces, k8sNamespace) && containsKey(StdoutIncludeSystemNamespaceSet, k8sNamespace) {
+				if len(StdoutIncludeSystemResourceSet) != 0 {
+					dsName, deploymentName := GetDSNameAndDeploymentNameFromK8sPodName(k8sPodName)
+					if !containsKey(StdoutIncludeSystemResourceSet, k8sNamespace+":"+dsName) && !containsKey(StdoutIncludeSystemResourceSet, k8sNamespace+":"+deploymentName) {
+						continue
+					}
+				}
+			}
 		} else if strings.EqualFold(logEntrySource, "stderr") {
 			if containerID == "" || containsKey(StderrIgnoreNsSet, k8sNamespace) {
 				continue
 			}
+			if containsKey(SystemNamespaces, k8sNamespace) && containsKey(StderrIncludeSystemNamespaceSet, k8sNamespace) {
+				if len(StderrIncludeSystemResourceSet) != 0 {
+					dsName, deploymentName := GetDSNameAndDeploymentNameFromK8sPodName(k8sPodName)
+					if !containsKey(StderrIncludeSystemResourceSet, k8sNamespace+":"+dsName) && !containsKey(StderrIncludeSystemResourceSet, k8sNamespace+":"+deploymentName) {
```

Comment:
We dont need to check both ds and deployment name. checking for controller name should consider all the possible controller types.